src/java/org/apache/cassandra/cql3/restrictions/SimpleRestriction.java

```diff
@@ -229,18 +231,42 @@ private List<ClusteringElements> bindAndGetClusteringElements(QueryOptions optio
         {
             case SINGLE_COLUMN:
             case TOKEN:
-                return bindAndGet(options).stream()
-                                          .map(b ->  ClusteringElements.of(columnsExpression.columnSpecification(), b))
-                                          .collect(Collectors.toList());
+                return bindAndGetSingleTermClusteringElements(options);
             case MULTI_COLUMN:
-                return bindAndGetElements(options).stream()
-                                                  .map(buffers -> ClusteringElements.of(columnsExpression.columns(), buffers))
-                                                  .collect(Collectors.toList());
+                return bindAndGetMultiTermClusteringElements(options);
             default:
                 throw new UnsupportedOperationException();
         }
     }
 
+    private List<ClusteringElements> bindAndGetSingleTermClusteringElements(QueryOptions options)
+    {
+        List<ByteBuffer> values = bindAndGet(options);
+        if (!values.isEmpty()) {
+            List<ClusteringElements> elements = new ArrayList<>(values.size());
+            for (ByteBuffer value : values) {
+                ClusteringElements byteBuffers = ClusteringElements.of(columnsExpression.columnSpecification(), value);
+                elements.add(byteBuffers);
```

Comment:
you could omit byteBuffers variable by putting the result of ClusteringElements.off directly to elements.add

By doing so, you will have a for loop consisting of one line of the code in its body., In that case you can omit braces for `for` entirely.